api/src/routes/donate.test.ts

```diff
@@ -0,0 +1,58 @@
+import request from 'supertest';
+
+import { setupServer, superRequest } from '../../jest.utils';
+
+describe('Donate', () => {
+  setupServer();
+
+  describe('Authenticated User', () => {
+    let setCookies: string[];
+
+    // Authenticate user
+    beforeAll(async () => {
+      const res = await request(fastifyTestInstance.server).get(
+        '/auth/dev-callback'
+      );
+      setCookies = res.get('Set-Cookie');
+    });
+
+    describe('POST /donate/add-donation', () => {
+      it('should return 200 and update the user', async () => {
+        const response = await superRequest('/donate/add-donation', {
+          method: 'POST',
+          setCookies
+        }).send({
+          isDonating: true
+        });
+
+        expect(response.status).toBe(200);
+        expect(response.body).toEqual({
+          isDonating: true
+        });
+      });
+
+      it('should return 400 if the body is empty', async () => {
+        const response = await superRequest('/donate/add-donation', {
+          method: 'POST',
+          setCookies
+        }).send({});
+
+        expect(response.status).toBe(400);
```

Comment:
```suggestion
      it('should return 400 if the user is already donating', async () => {
        const successResponse = await superRequest('/donate/add-donation', {
          method: 'POST',
          setCookies
        }).send({});
        
        expect(successResponse.status).toBe(200);       
        
        const failResponse = await superRequest('/donate/add-donation', {
          method: 'POST',
          setCookies
        }).send({});
    
        expect(failResponse.status).toBe(400);
```
Currently the donation fail with a 500, but 400 is more appropriate.

By the way for this to work, you'll need a `beforeEach` to reset the user's `isDonating` flag.

Reply:
That would be the case if there were more than 2 tests.

Reply:
Two tests are enough. The first test will set `isDonating` to `true` and the second test will fail as a result.